tests/test_sycl_queue.py

```diff
@@ -1245,3 +1245,13 @@ def test_indices(device):
 
     assert_allclose(numpy_array, dpnp_array)
     assert dpnp_array.sycl_device == device
+
+
+@pytest.mark.parametrize(
+    "device",
+    valid_devices,
+    ids=[device.filter_string for device in valid_devices],
+)
+def test_grid(device):
+    assert dpnp.mgrid_device(device=device)[0:4].sycl_device == device
```

Comment:
one more additional test scenario might be to create a SYCL queue from device and to pass to a function call:
```python

sycl_queue=dpctl.SyclQueue(device)
x = getattr(dpnp, func)(sycl_queue=sycl_queue)[0:4]
assert_sycl_queue_equal(x.sycl_queue, sycl_queue)
```